slither/tools/read_storage/__main__.py

```diff
@@ -157,6 +212,13 @@ def main() -> None:
         assert args.rpc_url
         srs.walk_slot_info(srs.get_slot_values)
 
+    if args.diff_block:
+        slots_block_orig = copy.deepcopy(srs.slot_info)
+        srs.block = diff_block
```

Comment:
This doesn't account for dynamic arrays (that would require calling `get_storage_layout` again) because the old block would have been used https://github.com/crytic/slither/blob/7ea572723b4eddb18e716c4ae94e0d2100791127/slither/tools/read_storage/read_storage.py#L602-L622

Reply:
Thanks for the info. I fixed it assuming that `get_all_storage_variables` should not be called again, as I understand that storage layout won't change. Is that correct?

By chance, do you know of any verified contract that uses dynamic arrays, to test against it?